### PR TITLE
Duke Nukies & stamcrit resistance

### PR DESCRIPTION
**Pull request by whateverusername0**
## About the PR

CL :trollface:
also added stamina damage resistance which can be enabled by simply adding a StaminaDamageResistance component with the coefficient to an entity. very cool
now all hardsuits have 25% stamdmg resistance, bloodred has 50%, elite - 75%, jugger - 65%
might add more stamdmg res to death squad or chrono gamer gear

## Why / Balance

i always wanted to balance nukies and i think i did a good job at it
GOD I LOVE TEH DUKE NUKEI

## Media
hardsuit stamcrit resistance check
image
image

roundstart nukie changes
image

cool new sprites
image

reinforcement changes
image
image

:cl:
- add: Now all hardsuits have different stamina damage resistance.
- add: Nuclear operatives now spawn without hardsuits but with 50 more TC.
- add: Nuclear operative reinforcements now spawn with a bonus uplink with 50 TC.
- tweak: Changed nuclear operative hardsuit sprites.
- tweak: Decreased raid suit price to 30 TC
- tweak: Increased syndicate elite suit price to 70 TC
- tweak: Increased juggernaut suit price to 100 TC
- tweak: Increased the chinalake price to 150 TC
- tweak: Decreased nukie reinforcement agent price to 125 TC

### CODE REVIEW

**Review by misandrie — CHANGES_REQUESTED** [review]
Hysterically overtuned balance tied to a resprite for no real reason - by killing stuns entirely you remove any counter to juggernaut players, even if jugg walks in solo into 3 secoffs with stunbatons at point blank range, which should be an instant over for the player due to severe skill issue.

The real killer are tranq and rubber bullets, which do absurd amounts of stamina damage at range and thus - at little risk.

Discussion on stamres was held here in march (wizden server):
https://discord.com/channels/310555209753690112/1217867633588830239/1217867633588830239

Taking away hardsuits but giving 50 more telecrystals effectively gives them 50 tc for free if they decide to take anything but the default hardsuit - the redundancy is a balance and noobtrap issue of some bright individuals forgetting to buy a hardsuit in the first place and getting forced into an EVA suit off the shuttle instead.

Movement speed increases are not mentioned anywhere.

**Comment by a-person5660** [comment]
i think that stun resist is way too high, 0% for civ suits, 25% base for sec and nukie suits, then maybe 50% for elite and jug. also make the elite suit have 0 speed penalty.

File: Resources/Locale/en-US/ghost/roles/ghost-role-component.ftl [hunk 1]
```diff
@@ -206,7 +206,7 @@ ghost-role-information-syndicate-reinforcement-spy-description = Someone needs r
 ghost-role-information-syndicate-reinforcement-thief-name = Syndicate Thief
 ghost-role-information-syndicate-reinforcement-thief-description = Someone needs reinforcements. Your job is to break in and retrieve something valuable for your agent.
 
-ghost-role-information-nukeop-reinforcement-name = Nuclear Operative
+ghost-role-information-nukeop-reinforcement-name = Nuclear Assailant
```
Comment (Piras314) [inline Resources/Locale/en-US/ghost/roles/ghost-role-component.ftl:209]: assailant doesn't make any sense here, put it back to op and if you really hate op I'd rather reinforcement than assailant

File: Resources/Prototypes/Entities/Clothing/OuterClothing/base_clothingouter.yml [hunk 2]
```diff
@@ -137,6 +137,8 @@
     - HidesHarpyWings # Goobstation
   - type: ClothingRequiredStepTriggerImmune
     slots: WITHOUT_POCKET
+  - type: StaminaDamageResistance # goobstation
+    coefficient: 0.75
```
Comment (Piras314) [inline Resources/Prototypes/Entities/Clothing/OuterClothing/base_clothingouter.yml:141]: remove this entirely

File: Resources/Prototypes/Entities/Clothing/OuterClothing/hardsuits.yml [hunk 3]
```diff
@@ -512,6 +512,8 @@
     - MonkeyWearable
     - Hardsuit
     - WhitelistChameleon
+  - type: StaminaDamageResistance # goobstation
+    coefficient: 0.5 # 50% resistance
```
Comment (Piras314) [inline Resources/Prototypes/Entities/Clothing/OuterClothing/hardsuits.yml:516]: 0.8

File: Resources/Prototypes/Entities/Clothing/OuterClothing/hardsuits.yml [hunk 4]
```diff
@@ -562,12 +564,14 @@
...
-  - type: ClothingSpeedModifier
+  - type: ClothingSpeedModifier # goobstation - duke nukies
     walkModifier: 1.0
-    sprintModifier: 0.9
+    sprintModifier: 1.0
   - type: HeldSpeedModifier
   - type: ToggleableClothing
     clothingPrototype: ClothingHeadHelmetHardsuitSyndieElite
+  - type: StaminaDamageResistance # goobstation
+    coefficient: 0.25 # 75% stamdamage resistance
```
Comment (Piras314) [inline Resources/Prototypes/Entities/Clothing/OuterClothing/hardsuits.yml:574]: 0.8

File: Resources/Prototypes/Entities/Clothing/OuterClothing/hardsuits.yml [hunk 5]
```diff
@@ -627,11 +631,13 @@
...
   - type: ClothingSpeedModifier
-    walkModifier: 0.9
-    sprintModifier: 0.65
+    walkModifier: 0.7 # goobstation - duke nukies
+    sprintModifier: 0.6
   - type: HeldSpeedModifier
   - type: ToggleableClothing
     clothingPrototype: ClothingHeadHelmetHardsuitCybersun
+  - type: StaminaDamageResistance # goobstation
+    coefficient: 0.35 # 65% stamdmg resist
```
Comment (Piras314) [inline Resources/Prototypes/Entities/Clothing/OuterClothing/hardsuits.yml:640]: 0.65 and put the speed back up

**Review by Piras314 — CHANGES_REQUESTED** [review]
Revert all the TC and reinforcement changes.

Also see above for balance suggestions

**Comment by misandrie** [comment]
> OP doesn't explain a single bit about balancing
> Maintainer jr. doesn't explain anything about his requested balance changes in response

**Comment by a-person5660** [comment]
also, you should probably make the resprite a seperate PR from this.

**Comment by Piras314** [comment]
just initial nerfs to get it in some kind of reasonable ballpark, will calculate the required number of stuns etc in a bit

**Comment by misandrie** [comment]
> just initial nerfs to get it in some kind of reasonable ballpark, will calculate the required number of stuns etc in a bit

There's not a single effective nerf in this PR.

image

**Comment by Piras314** [comment]
being immune to stuns with stimulants is fine

**Comment by misandrie** [comment]
> being immune to stuns with stimulants is fine

You're expected to elaborate

**Comment by Piras314** [comment]
stims making you stun immune is fine, they're not easy to get en masse and you have to constantly inject yourself

**Comment by misandrie** [comment]
> stims making you stun immune is fine, they're not easy to get en masse and you have to constantly inject yourself

I don't believe you've seen HM2 Tony RP with a guy full of stimkits and northstars getting full combo on the entire station before. Stimulants do more than just double your total stamina atm - they also massively increase your speed and heal you when you're at 70 damage or above but not crit.

There are already questionable combos balance-wise with the stimulants (the fucking desword) as-is, and this is a direct buff to hardsuits over a web vest (where 20% pierce for 50%/75% stam is the best tradeoff in the game atm).

**Comment by misandrie** [comment]
This is not to mention that you'll be playing with abstract numbers and balance ping-pong because you'll have to now account for the stam resist modifiers and the total amount of stamina you have.

**Comment by whateverusername0** [comment]
ok addressing changes now

> tied to a resprite for no real reason

my bad og, it was supposed to be a modsuit PR but i got a bit carried away

> Taking away hardsuits but giving 50 more telecrystals effectively gives them 50 tc for free if they decide to take anything but the default hardsuit

ok, so what? why not eco and buy something like raid suits? this leaves more TC for nukies to spend on gimmicks

> bright individuals forgetting to buy a hardsuit in the first place and getting forced into an EVA suit off the shuttle instead

if someone doesn't buy armor that's his problem, why do we need to tune the mechanics towards the new playerbase? they won't be able to learn from mistakes if we remove mistakes from the game :trollface:

File: Resources/Prototypes/Entities/Clothing/OuterClothing/base_clothingouter.yml (hunk 2 above)
Comment (whateverusername0) [reply Resources/Prototypes/Entities/Clothing/OuterClothing/base_clothingouter.yml:141]: don't see an issue here, taking one more hit from a batong to go stamcrit while wearing a hardsuit should be nice

File: Resources/Prototypes/Entities/Clothing/OuterClothing/hardsuits.yml (hunk 4 above)
Comment (whateverusername0) [reply Resources/Prototypes/Entities/Clothing/OuterClothing/hardsuits.yml:574]: please give elite some love hot damn
it's less resistant to brute therefore should at least have more stamresist

**Comment by whateverusername0** [comment]
ok here's my stam resistance changes:
all nukie hardsuits - 50%
engi, security - 25%
warden, CE - 35%
hos, captain, wizard, ERT - 50%
death squad - 90%

**Comment by misandrie** [comment]
> if someone doesn't buy armor that's his problem, why do we need to tune the mechanics towards the new playerbase? they won't be able to learn from mistakes if we remove mistakes from the game :trollface:

Mistakes of one guy reflect on the whole team. If this skill issue wouldn’t have affected anyone else you’d have some ground to stand on. People are already at each other’s throats for minor mistakes when playing ops.

**Comment by jorgun** [comment]
> ok here's my stam resistance changes: all nukie hardsuits - 50% engi, security - 25% warden, CE - 35% hos, captain, wizard, ERT - 50% death squad - 90%

this sounds more resonabe, we shud to a trila weekend to see how this works

**Comment by whateverusername0** [comment]
i renamed the branch. well shit

**Pull request closed without merging**